### PR TITLE
docs: Add balloon documentation

### LINKED ISSUE #3807: docs: Add documentation for `balloon`
We need to add some documentation that will explain the different options the user can enable through the `balloon` device, and the purpose for each of them.

### PR DESCRIPTION
**Pull request by sboeuf**
Document the reason why someone might need to use the balloon device and
detail the different parameters associated with it.

Fixes #3807

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>

### CODE REVIEW

File: docs/balloon.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Balloon
+
+Cloud Hypervisor emulates a balloon device based on the VIRTIO specification.
```
Comment (rbradford) [inline docs/balloon.md:3]: There is no emulation here. This is a paravirtualised device and no such physical device type exists.

File: docs/balloon.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+```
+
+```
+--balloon <balloon>	Balloon parameters "size=<balloon_size>,deflate_on_oom=on|off,free_page_reporting=on|off"
+```
+
+### `size`
+
+Size of the balloon device. It is substracted from the VM's total size. For
```
Comment (rbradford) [inline docs/balloon.md:27]: nit: Typo.

File: docs/balloon.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+instance, if creating a VM with 4GiB of RAM, along with a balloon of 1GiB, the
+guest will find 3GiB of accessible memory.
```
Comment (rbradford) [inline docs/balloon.md:29]: s/find 3GiB/will be able to use 3GiB/

The guest sees all the RAM an unless it is balloon enlightened is entitled to all of it.

File: docs/balloon.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+This parameter is mandatory.
+
+Value is an unsigned integer of 64 bits.
```
Comment (rbradford) [inline docs/balloon.md:33]: Probably want to say that it is the size of bytes.

File: docs/balloon.md (hunk 1 above)
Comment (sboeuf) [reply docs/balloon.md:3]: Yes definitely wrong wording. Do you want me to replace `emulates` with `implements`?

File: docs/balloon.md (hunk 2 above)
Comment (sboeuf) [reply docs/balloon.md:27]: Where's the typo?

Comment (rbradford) [reply docs/balloon.md:27]: > Where's the typo?

"substracted" ?

Comment (sboeuf) [reply docs/balloon.md:27]: Well I learned something today, I thought `subtract` was written `substract` :rofl:

**Review by rbradford — APPROVED** [review]

**Pull request merged**